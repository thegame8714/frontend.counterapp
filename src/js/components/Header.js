import React from "react";
import { Link } from 'react-router';

class Header extends React.Component {
	render() {
		return (
			<header className="bar bar-nav">
				<Link className="icon icon-left-nav pull-left" to="/"></Link>
				<Link className="icon icon-gear pull-right" to="/sync"></Link>
				<h1 className="title">
					<Link className="navbar-brand" to="/">CounterApp</Link>
				</h1>
			</header>
		)
	}
}

export default Header;
