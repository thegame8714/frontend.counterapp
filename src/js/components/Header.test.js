import React from 'react';
import { mount, shallow } from 'enzyme';
import { mountToJson } from 'enzyme-to-json'
import Header from './Header';

describe('<Header />', () => {
  it('should match snapshot', () => {
    const wrapper = mount(<Header />);
    expect(mountToJson(wrapper)).toMatchSnapshot();
  })
  it('should have one header tag', () => {
    const wrapper = mount(<Header />);
    expect(wrapper.find('header').length).toBe(1);
  })
  it('should have bar class on header tag', () => {
    const wrapper = mount(<Header />);
    expect(wrapper.find('header').hasClass('bar')).toBe(true);
  });
  it('should have three Links', () => {
    const wrapper = mount(<Header />);
    expect(wrapper.find('Link').length).toBe(3);
  });
  it('should have one H1', () => {
    const wrapper = mount(<Header />);
    expect(wrapper.find('h1').length).toBe(1);
  });
  it('should have CounterApp in H1', () => {
    const wrapper = mount(<Header />);
    expect(wrapper.find('h1').text()).toBe('CounterApp');
  });

});