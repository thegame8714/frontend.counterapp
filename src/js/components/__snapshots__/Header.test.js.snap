exports[`<Header /> should match snapshot 1`] = `
<Header>
  <header
    className="bar bar-nav">
    <Link
      className="icon icon-left-nav pull-left"
      onlyActiveOnIndex={false}
      style={Object {}}
      to="/">
      <a
        className="icon icon-left-nav pull-left"
        onClick={[Function]}
        style={Object {}} />
    </Link>
    <Link
      className="icon icon-gear pull-right"
      onlyActiveOnIndex={false}
      style={Object {}}
      to="/sync">
      <a
        className="icon icon-gear pull-right"
        onClick={[Function]}
        style={Object {}} />
    </Link>
    <h1
      className="title">
      <Link
        className="navbar-brand"
        onlyActiveOnIndex={false}
        style={Object {}}
        to="/">
        <a
          className="navbar-brand"
          onClick={[Function]}
          style={Object {}}>
          CounterApp
        </a>
      </Link>
    </h1>
  </header>
</Header>
`;
